clang/lib/CIR/CodeGen/CIRGenBuiltinAArch64.cpp

```diff
@@ -2332,6 +2332,17 @@ mlir::Value CIRGenFunction::buildCommonNeonBuiltinExpr(
     return builder.create<mlir::cir::ShiftOp>(
         loc, vTy, builder.createBitcast(ops[0], vTy), ops[1], true);
   }
+  case NEON::BI__builtin_neon_vtst_v:
+  case NEON::BI__builtin_neon_vtstq_v: {
+    mlir::Location loc = getLoc(e->getExprLoc());
+    ops[0] = builder.createBitcast(ops[0], ty);
+    ops[1] = builder.createBitcast(ops[1], ty);
+    ops[0] = builder.createAnd(ops[0], ops[1]);
+    // CIR VecCmpO LLVM Lowering always creates SExt for vector of int type
```

Comment:
VecCmpO -> VecCmpOp

Reply:
LLVM isn't the only path to lower CIR from nor we should assume the reader knows anything about OG codegen. It's fine if you add a note about it but make sure you are not making assumptions of what this is going to be lowered to. Example:
```
// Note that during LLVM lowering `VecCmpOp` result is sign extended, matching what traditional codegen behavior.
```

Reply:
makes sense. Especially without looking at OG code, reader wouldn't know why this comment is here. Will change it.